### DISCUSSION TITLE
Feature Request: Debug Mode

### DISCUSSION THREAD

**Discussion by ghost** [colinwilson/lotusdocs#40] [Feature Requests]
The current light and dark theme have *glossy* light (white) and dark (black) backgrounds. 

A debug mode which has lower contrast will improve accessibility. Hugo themes like [Hermit][1] and [Call Me Sam][2] are extremely comfortable on the eyes when working for prolonged sessions and are very easy to read. Colorshemes like [Gruvbox][4] and [Solarized][5] also come to mind - both implement light and dark background which are more accessible.

As for the debug information:

+ Pages which don't have a last updated, could have that information (homepage and nested lists like [this page](https://lotusdocs.dev/docs/guides/features/)).

+ The last updated information like "Last updated 3 days ago" [here][3], could point to the Git commit hash

+ The "ON THIS PAGE" section could have a slightly different contrast than the rest of the content.

[1]: https://hugo-theme-hermit.netlify.app/
[2]: https://victoria.dev/hugo-theme-sam/about/
[3]: https://lotusdocs.dev/docs/overview/
[4]: https://github.com/morhetz/gruvbox
[5]: https://ethanschoonover.com/solarized/

**Comment by colinwilson** [comment]
This is an interesting idea. I was already considering a 'debug' / 'editing` mode for working locally. I'll likely incorporate some of your suggestions. Changing the background and text colours for working locally shouldn't be too difficult.


> Pages which don't have a last updated, could have that information (homepage and nested lists like [this page](https://lotusdocs.dev/docs/guides/features/)).

Not sure what you mean by this. Could you explain a little be more?

> The last updated information like "Last updated 3 days ago" [here](https://lotusdocs.dev/docs/overview/), could point to the Git commit hash

Do you mean like the `edit this page` link?

> The "ON THIS PAGE" section could have a slightly different contrast than the rest of the content.

In mobile or desktop view? or both?

**Reply by ghost** [reply]
> > Pages which don't have a last updated, could have that information (homepage and nested lists like [this page](https://lotusdocs.dev/docs/guides/features/)).

> Not sure what you mean by this. Could you explain a little be more?

The homepage too could indicate when the last addition or change was made to any page on the website. For example, if I update the [overview page](https://lotusdocs.dev/docs/overview/), or add a new page, at `YYYY-MM-DD HH:MM`, it could be shown on the homepage in the bottom right corner.

Pages containing a list, like the [guides page](https://lotusdocs.dev/docs/guides/), and pages containing a nested list, like the [features page](https://lotusdocs.dev/docs/guides/features/) dont have a "Last updated" that single pages like [syntax highlighting](https://lotusdocs.dev/docs/guides/features/syntax-highlighting/) have at the bottom right.  

These list and nested list pages could contain the "Last modified" at the bottom right. And like the homepage, these list pages could show this information when a page is added or modified within it.

So in summary, the homepage could contain "Last modified" based on site-wide changes (addition or modification), and lists and nested lists could contain "Last modified" based on changes made within that particular section.

---

> > The last updated information like "Last updated 3 days ago" [here](https://lotusdocs.dev/docs/overview/), could point to the Git commit hash

> Do you mean like the `edit this page` link?

I mean the Git commit hash pointing to the commit in repository. Take the [bare-min-theme](https://hugo-sandbox.netlify.app) for instance. Above the image there's a git commit hash [cf4db09](https://gitlab.com/kaushalmodi/hugo-sandbox/commit/cf4db0990d6cd3e445896669d55da1223a9a8c5e). If you click it, you will be redirected to the source repository, and will be shown all the files which were added or modified in that particular commit.

---

> > The "ON THIS PAGE" section could have a slightly different contrast than the rest of the content.

> In mobile or desktop view? or both?

I think debug mode makes more sense as a desktop only option (at least for the proposals I've made).
